Docker/validate.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python
+"""Validation script for scRNA-seq signal correction.
+Predictions file must be a zipped tarball archive of imputed count files:
+(*.tar.gz).
+Each imputed count file must have a correct file format:
+(pac_expId_ds_prop_imputed.csv).
+"""
+
+import argparse
+import json
+import pandas as pd
+from zipfile import ZipFile
+
+
+def get_args():
+    """Set up command-line interface and get arguments."""
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-r", "--results", required=True,
+                        help="validation results")
+    parser.add_argument("-e", "--entity_type", required=True,
+                        help="synapse entity type downloaded")
+    parser.add_argument("-s", "--submission_file", help="Submission File")
+    parser.add_argument("-g", "--goldstandard", required=True,
+                        help="Goldstandard for scoring")
+    return parser.parse_args()
+
+
+def get_dim_names(df):
+    """
+    Get all row names and column names of df
+    """
+    out = {"rownames": df.index.tolist(),
+           "colnames": df.columns.values.tolist()}
+    return out
+
+
+def main():
+    """Main function."""
+    args = get_args()
+
+    invalid_reasons = []
+    gs_file_status = True
+    prediction_file_status = True
+    exp_ids = ["2400", "2401", "7200", "7201"]
+    ds_props = ["0_125"]
+
+    # validate goldstandard file
+    if args.goldstandard is None:
+        gs_file_status = False
+        invalid_reasons = [args.goldstandard + ' not found']
+    else:
+        gs_zip_file = ZipFile(args.goldstandard, "r")
```

Comment:
In Python, it is good practice to always explicitly close files you open (rather than relying on Python to do it). One of the biggest reasons why is that it'll release resources, e.g. RAM, which in turn, should improve performance!

You can either add `gs_zip_file.close()` after you're done with it or (generally preferred) use the `with` context manager:

```python
with zipfile.ZipFile(args.goldstandard) as gs_zip_file:
   ...
```

In the latter approach, Python will automatically close the file for you once it leaves the code block.